llmclient/llms.py

```diff
@@ -596,12 +598,22 @@ async def acompletion(self, messages: list[Message], **kwargs) -> list[LLMResult
                 output_messages = [Message(**completion.message.model_dump())]
 
             reasoning_content = None
-            if hasattr(completion.message, "provider_specific_fields"):
+            if (
+                hasattr(completion.message, "provider_specific_fields")
+                and completion.message.provider_specific_fields is not None
+            ):
+                # DeepSeek's reasoning:
+                # https://github.com/BerriAI/litellm/issues/7877
                 provider_specific_fields = completion.message.provider_specific_fields
                 if isinstance(provider_specific_fields, dict):
                     reasoning_content = provider_specific_fields.get(
-                        "reasoning_content", None
+                        "reasoning_content"
                     )
+            elif hasattr(completion.message, "reasoning"):
+                # OpenRouter's reasoning:
```

Comment:
```suggestion
                # NOTE: OpenRouter's reasoning was added in:
```

Same here, I was not sure why you linked a closed issue, let's just give more context